libs/language-server/src/lib/validation/checks/valuetype-definition.spec.ts

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { AstNode, AstNodeLocator, LangiumDocument } from 'langium';
+import { NodeFileSystem } from 'langium/node';
+
+import {
+  ValidationContext,
+  ValuetypeDefinition,
+  createJayveeServices,
+  useExtension,
+} from '../../../lib';
+import {
+  ParseHelperOptions,
+  expectNoParserAndLexerErrors,
+  parseHelper,
+  readJvTestAssetHelper,
+  validationAcceptorMockImpl,
+} from '../../../test';
+import { TestLangExtension } from '../../../test/extension';
+
+import { validateValuetypeDefinition } from './valuetype-definition';
+
+describe('valuetype-definition validation tests', () => {
+  let parse: (
+    input: string,
+    options?: ParseHelperOptions,
+  ) => Promise<LangiumDocument<AstNode>>;
+
+  const validationAcceptorMock = jest.fn(validationAcceptorMockImpl);
+
+  let locator: AstNodeLocator;
+
+  const readJvTestAsset = readJvTestAssetHelper(
+    __dirname,
+    '../../../test/assets/',
+  );
+
+  async function parseAndValidateValuetypeDefinition(input: string) {
+    const document = await parse(input);
+    expectNoParserAndLexerErrors(document);
+
+    const valuetypeDefinition = locator.getAstNode<ValuetypeDefinition>(
+      document.parseResult.value,
+      'valuetypes@0',
+    ) as ValuetypeDefinition;
+
+    validateValuetypeDefinition(
+      valuetypeDefinition,
+      new ValidationContext(validationAcceptorMock),
+    );
+  }
+
+  beforeAll(() => {
+    // Register test extension
+    useExtension(new TestLangExtension());
+    // Create language services
+    const services = createJayveeServices(NodeFileSystem).Jayvee;
+    locator = services.workspace.AstNodeLocator;
+    // Parse function for Jayvee (without validation)
+    parse = parseHelper(services);
+  });
+
+  afterEach(() => {
+    // Reset mock
+    validationAcceptorMock.mockReset();
+  });
+
+  it('should have no error on empty constraint list', async () => {
+    const text = readJvTestAsset(
+      'valuetype-definition/valid-valuetype-definition.jv',
+    );
+
+    await parseAndValidateValuetypeDefinition(text);
+
+    expect(validationAcceptorMock).toHaveBeenCalledTimes(0);
+  });
+
+  it('error on supertype cycle', async () => {
+    const text = readJvTestAsset(
+      'valuetype-definition/invalid-supertype-cycle.jv',
+    );
+
+    await parseAndValidateValuetypeDefinition(text);
+
+    // TODO FIND OUT HOW THAT WORKS
```

Comment:
I guess you already found out how to do it, right? Otherwise, feel free to ask for help.

Reply:
Yeah found out, forgot to remove TODO :sweat_smile: